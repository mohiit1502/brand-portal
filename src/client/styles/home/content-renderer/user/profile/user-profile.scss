@import "../../../../override/variables";

.user-profile-content {
  .content-header-row {
    font-weight: bold;
  }
  .password-reset-col{
    button{
      border-radius: 42px;
      font-weight: bold;
      font-size: 15px;

      &.reset-password {
        border: $default-input-border;
        color: $market-place-blue;
      }

      &.manage-profile {
        color: white;
      }

      &:hover {
        color: white;
        background: $market-place-blue;
      }
    }
  }

  .edit-button-panel {
    .btn-primary{
      border-radius: 42px;
      background: #006799;
      font-weight: bold;
      font-size: 15px;
      border: none;
    }
    .btn-link{
      font-weight: bold;
      font-size: 18px;
      text-decoration: underline;
    }
  }
  .password-reset-row {
    .badge {
      //border: 1px solid $blue;
    }
  }
}
