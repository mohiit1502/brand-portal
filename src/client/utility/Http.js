/* eslint-disable complexity */
/* eslint-disable no-unused-expressions */
/* eslint-disable max-params */
/* eslint-disable max-statements */
import fetch from "node-fetch";
import queryString from "query-string";
import ClientHttpError from "./ClientHttpError";
import { NOTIFICATION_TYPE } from "../actions/notification/notification-actions";
import CONSTANTS from "../constants/constants";
export default class Http {

  static async get(url, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});

    const options = {
      method: "GET"
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;

    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message, err.code);
  }

  static async post(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "POST",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json"
      }
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);

  }

  static async postAsFormData(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "POST",
      body: data,
      headers: { }
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);

  }

  static async put(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "PUT",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json"
      }
    };

    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);
  }

  static async delete(url, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "DELETE",
      noInject: true
    };
    return fetch(urlString, options);
  }

  displayToast(status, toastcallback) {
    if (new RegExp(CONSTANTS.CODES.ERRORCODES.SERVERERROR).test(status)) {
      toastcallback && typeof toastcallback === "function" && toastcallback(NOTIFICATION_TYPE.ERROR, `Unable to reach our services currently, please try again in sometime.`);
    } else if (new RegExp(CONSTANTS.CODES.ERRORCODES.FOURNOTFOUR).test(status)) {
      toastcallback && typeof toastcallback === "function" && toastcallback(NOTIFICATION_TYPE.ERROR, `Requested Resource could not found.`);
    }
  }
}
