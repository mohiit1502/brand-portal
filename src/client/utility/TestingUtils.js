import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const realStore = state => {
  const configureStore = initialState => {
    const composeEnhancers =
      typeof window === "object" &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          serialize: { // prettier-ignore
            immutable: Immutable
          }
        }) : compose;
    const store = createStore(rootReducer, initialState, composeEnhancers(applyMiddleware(thunk)));

    if (module.hot) {
      module.hot.accept("./reducers", () => {
        const nextRootReducer = require("./reducers").default;
        store.replaceReducer(nextRootReducer);
      });
    }

    return store;
  };

  return configureStore(state || window.__WML_REDUX_INITIAL_STATE__);
}

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
