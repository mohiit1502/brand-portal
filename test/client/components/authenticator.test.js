import React from "react";
import {Provider} from "react-redux";
import toJson from "enzyme-to-json";
import {BrowserRouter} from "react-router-dom";
import Cookies from "electrode-cookies";
import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Authenticator from "../../../src/client/components/authenticator";
import Http from "../../../src/client/utility/Http";
import {setupFetchStub, testStore} from "../../../src/client/utility/TestingUtils";
import profile from "../mocks/userProfile";
import modalsMeta from "../../../src/client/config/modals-meta";

configure({ adapter: new Adapter() });
jest.mock("electrode-cookies");

const setUp = () => {
  const mockStore = {
    company: {isNew: true},
    content: {metadata: modalsMeta},
    user: {profile}
  };
  const props = {
    location: {pathname: "/dashboard"}
  }
  return mount(<Provider store={testStore(mockStore)}>
      <BrowserRouter>
          <Authenticator {...props} />
      </BrowserRouter>
    </Provider>);
};

describe("Tests for redirection logic in Authenticator", () => {
  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });

  it("renders authenticator without error", () => {
    Cookies.get.mockImplementation(() => "test-cookie");
    let getter = jest.spyOn(Http, "get").mockImplementation(url => {
      switch (url) {
        case '/api/claims':
          return Promise.resolve({
            body: {
              content: [
                {
                  firstName: "test-f-name",
                  lastName: "test-l-name",
                  email: "abc@gmail.com",
                  brands: [],
                  status: "ACTIVE",
                  role: {name: "test-role"},
                  type: "thirdparty"
                }
              ]
            }
          })
        case '/api/brands':
        case '/api/users':
          return Promise.resolve({
            body: {
              content: [
                {
                  firstName: "test-f-name",
                  lastName: "test-l-name",
                  email: "abc@gmail.com",
                  brands: [],
                  status: "ACTIVE",
                  role: {name: "test-role"},
                  type: "thirdparty"
                }
              ]
            }
          })
        default:
          return Promise.resolve({body: "{}"});
      }
    });
    global.fetch = jest.fn().mockImplementation(setupFetchStub({}));
    wrapper = setUp();
    let tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    global.fetch.mockClear();
  });


});
