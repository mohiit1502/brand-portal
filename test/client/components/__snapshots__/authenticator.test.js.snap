// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for redirection logic in Authenticator renders authenticator without error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Authenticator)
        location={
          Object {
            "pathname": "/dashboard",
          }
        }
      >
        <Authenticator
          dispatchBrands={[Function]}
          dispatchClaims={[Function]}
          dispatchFormFieldMetadata={[Function]}
          dispatchLogoutUrl={[Function]}
          dispatchModalsMetadata={[Function]}
          dispatchSectionsMetadata={[Function]}
          dispatchUsers={[Function]}
          isNew={true}
          location={
            Object {
              "pathname": "/dashboard",
            }
          }
          modalsMeta={
            Object {
              "ACCOUNT_LINKED": Object {
                "BODY_CONTENT_CLASSES": "margin-top-title",
                "HEIGHT": 80,
                "IMAGE": "SuccessCheck",
                "MESSAGE": Object {
                  "classes": "mt-4",
                  "content": Object {
                    "para1": Object {
                      "classes": "subtitle mx-5",
                      "text": "Success! Your account has been successfully updated.",
                    },
                    "para2": Object {
                      "classes": "subtitle mx-5",
                      "text": "Going forward, please use your Seller Center login credentials to access your Brand Portal account.",
                    },
                  },
                },
                "PRIMARY_ACTION": Object {
                  "action": "toNext",
                  "text": "Continue to my account",
                },
                "TITLE": Object {
                  "classes": "justify-content-center",
                  "content": Object {
                    "para": Object {
                      "text": "Success!",
                    },
                  },
                },
                "TYPE": "CTA",
              },
              "ACCOUNT_LINKING": Object {
                "ADDITIONAL_ACTION": Object {
                  "action": "logout",
                  "classes": "mp-blue",
                  "text": "Log out",
                },
                "CODE": 256,
                "IMAGE": "ShieldTick",
                "MESSAGE": Object {
                  "classes": "mt-4",
                  "content": Object {
                    "para1": Object {
                      "classes": "subtitle mx-4",
                      "text": "Confirm and connect your accounts to continue:",
                    },
                    "para2": Object {
                      "classes": "subtitle mx-4",
                      "text": "The Walmart Seller Center credentials you used to log in are associated with the company details existing in your Brand Portal account for:",
                    },
                    "para3": Object {
                      "classes": "font-weight-bold subtitle mt-2 mb-4 mx-4",
                      "dynamicReplacementConfig": Object {
                        "__namePlaceholder__": "profile.organization.name",
                      },
                      "text": "[__namePlaceholder__]",
                    },
                    "para4": Object {
                      "classes": "subtitle mx-3",
                      "text": "In order to access the Brand Portal dashboard, you must connect your accounts to override your initial Brand Portal login credentials.",
                    },
                  },
                },
                "PRIMARY_ACTION": Object {
                  "action": "linkAccounts",
                  "text": "Connect accounts",
                },
                "TITLE": Object {
                  "classes": "justify-content-center",
                  "content": Object {
                    "para": Object {
                      "text": "Link your accounts to continue",
                    },
                  },
                },
                "TYPE": "CTA",
              },
              "ACCOUNT_LINKING_CONFIRM": Object {
                "ADDITIONAL_ACTION": Object {
                  "action": "logout",
                  "classes": "mp-blue mt-3",
                  "text": "No, I will do this later",
                },
                "IMAGE": "ShieldTick",
                "MESSAGE": Object {
                  "classes": "mt-4",
                  "content": Object {
                    "para1": Object {
                      "classes": "subtitle mx-5",
                      "text": "Linking account needs only to occur once, but cannot be undone. If you decide to do this later, you will be logged out.",
                    },
                  },
                },
                "PRIMARY_ACTION": Object {
                  "action": "linkAccounts",
                  "text": "Yes, link my accounts",
                },
                "TITLE": "Are you sure?",
                "TYPE": "CTA",
              },
              "APPLICATION_INFORMATION": Object {
                "BODY_CLASSES": "px-5 modal-background modal-height",
                "BODY_CONTENT_CLASSES": "ml-5 text-left pl-5",
                "CODE": Array [
                  2,
                  8,
                ],
                "CONTENT_COMPONENT": "ApplicationDetails",
                "CONTENT_COMPONENT_PROP": "updateCompanyDetails",
                "MESSAGE": Object {
                  "classes": "",
                  "content": Object {
                    "para2": Object {
                      "classes": "mt-3",
                      "text": "",
                    },
                    "para3": Object {
                      "classes": "mt-2",
                      "renderCondition": "{\\"keyPath\\": \\"orgStatus\\", \\"value\\": \\"on_hold\\"}",
                      "text": "Review and confirm that the information in your application is accurate.",
                    },
                    "para4": Object {
                      "classes": "border-bottom my-3",
                      "text": "",
                    },
                    "partial1": Object {
                      "chunk1": "Status: ",
                      "chunk2": Array [
                        Object {
                          "classes": "status-review",
                          "key": Array [
                            "NEW",
                          ],
                          "text": "Under Review",
                        },
                        Object {
                          "classes": "status-onhold",
                          "key": Array [
                            "ON_HOLD",
                          ],
                          "text": "Action Required",
                        },
                        Object {
                          "classes": "status-declined",
                          "key": Array [
                            "REJECTED",
                            "REJECTED_ON_AUDIT",
                          ],
                          "text": "Declined",
                        },
                      ],
                    },
                    "partial2": Object {
                      "anchor": Array [
                        Object {
                          "classes": "font-weight-bold mt-5 brand-portal-link-class",
                          "href": "mailto:brandportal@walmart.com",
                          "key": Array [
                            "REJECTED",
                            "REJECTED_ON_AUDIT",
                          ],
                          "text": "brandportal@walmart.com.",
                        },
                      ],
                      "chunk1": Array [
                        Object {
                          "classes": "mt-5",
                          "key": Array [
                            "ON_HOLD",
                          ],
                          "text": "We have sent an email to ",
                        },
                        Object {
                          "classes": "",
                          "key": Array [
                            "NEW",
                          ],
                          "text": "Your company information and brand details are currently being reviewed. ",
                        },
                        Object {
                          "classes": "",
                          "key": Array [
                            "REJECTED",
                            "REJECTED_ON_AUDIT",
                          ],
                          "text": "Your application has been declined. We have sent an email to ",
                        },
                      ],
                      "chunk2": Array [
                        Object {
                          "classes": "font-weight-bold mt-5",
                          "dynamicReplacementConfig": Object {
                            "__emailPlaceholder__": "profile.email",
                          },
                          "key": Array [
                            "ON_HOLD",
                          ],
                          "text": "__emailPlaceholder__",
                        },
                        Object {
                          "classes": "",
                          "key": Array [
                            "NEW",
                          ],
                          "text": "Once your application is approved, you will receive a confirmation email at ",
                        },
                        Object {
                          "classes": "font-weight-bold mt-5",
                          "dynamicReplacementConfig": Object {
                            "__emailPlaceholder__": "profile.email",
                          },
                          "key": Array [
                            "REJECTED",
                            "REJECTED_ON_AUDIT",
                          ],
                          "text": "__emailPlaceholder__ ",
                        },
                      ],
                      "chunk3": Array [
                        Object {
                          "classes": "mt-5",
                          "dynamicReplacementConfig": Object {
                            "__emailPlaceholder__": "profile.email",
                          },
                          "key": Array [
                            "ON_HOLD",
                          ],
                          "text": " regarding your application. Please follow the instructions in the email and make any necessary changes to your application.",
                        },
                        Object {
                          "classes": "font-weight-bold mt-5",
                          "dynamicReplacementConfig": Object {
                            "__emailPlaceholder__": "profile.email",
                          },
                          "key": Array [
                            "NEW",
                          ],
                          "text": "__emailPlaceholder__.",
                        },
                      ],
                      "chunk4": Array [
                        Object {
                          "classes": "mt-5",
                          "dynamicReplacementConfig": Object {
                            "__emailPlaceholder__": "profile.email",
                          },
                          "key": Array [
                            "REJECTED",
                            "REJECTED_ON_AUDIT",
                          ],
                          "text": "regarding your application. If you believe this is an error, contact ",
                        },
                      ],
                    },
                    "partial3": Object {
                      "anchor": Array [
                        Object {
                          "classes": "font-weight-bold mt-5 brand-portal-link-class",
                          "href": "mailto:brandportal@walmart.com",
                          "key": Array [
                            "ON_HOLD",
                          ],
                          "text": "brandportal@walmart.com.",
                        },
                      ],
                      "chunk5": Array [
                        Object {
                          "classes": "mt-5",
                          "dynamicReplacementConfig": Object {
                            "__emailPlaceholder__": "profile.email",
                          },
                          "key": Array [
                            "ON_HOLD",
                          ],
                          "text": "If you believe this is an error, contact ",
                        },
                      ],
                    },
                  },
                },
                "MODAL_DIALOG_CLASSES": "width-unset modal-xl ",
                "TITLE": Object {
                  "classes": "col ml-3 pl-5",
                  "content": Object {
                    "button1": Object {
                      "buttonText": "Edit Application",
                      "classes": "col btn btn-primary px-0",
                      "icon": "EditOutlinedIcon",
                      "onClick": "dispatchUserProfile",
                      "renderCondition": "{\\"keyPath\\": \\"orgStatus\\", \\"value\\": \\"on_hold\\"}",
                    },
                    "para": Object {
                      "classes": "text-left h3 pl-0 font-weight-bold col-9",
                      "dynamicReplacementConfig": Object {
                        "__businessName__": "org.name",
                      },
                      "text": "__businessName__",
                    },
                  },
                },
                "TYPE": "CTA",
              },
              "APPLICATION_SUBMITTED": Object {
                "BODY_CLASSES": "px-5",
                "BODY_CONTENT_CLASSES": "text-left",
                "HEIGHT": 100,
                "IMAGE": "GreenCircleWhiteTick",
                "MESSAGE": Object {
                  "classes": "",
                  "content": Object {},
                },
                "MODAL_DIALOG_CLASSES": "width-unset",
                "PRIMARY_ACTION": Object {
                  "action": "refreshAndHideModal",
                  "text": "OK",
                },
                "SUBTITLE": Object {
                  "content": Object {
                    "para1": Object {
                      "classes": "font-size-16 mx-5",
                      "text": "You can check the status of your application anytime by logging into your Brand Portal account.",
                    },
                  },
                },
                "TITLE": Object {
                  "classes": "justify-content-center",
                  "content": Object {
                    "para": Object {
                      "classes": "h3 mx-5 px-5 mb-2 font-weight-bold",
                      "text": "Application Submitted",
                    },
                  },
                },
                "TYPE": "CTA",
              },
              "DASHBOARD_MODAL": Object {
                "ADDITIONAL_ACTION": Object {
                  "action": "closeModal",
                  "classes": "mt-0 d-inline-block text-right mr-3",
                  "text": "I'll do it later",
                },
                "BODY_CLASSES": "px-5",
                "BODY_CONTENT_CLASSES": "text-left px-5",
                "FOOTER_CLASSES": "margin-unset bg-blue mt-4",
                "HEADER": "Welcome to Walmart Brand Portal!",
                "IMAGE": "ProblemSolving",
                "MODAL_DIALOG_CLASSES": "width-unset modal-md",
                "PRIMARY_ACTION": Object {
                  "action": "goToUserProfile",
                  "actionParam": "https://seller.walmart.com",
                  "classes": "text-right float-right px-3",
                  "containerClasses": "p-3",
                  "text": "Go to your Profile",
                },
                "SUBTITLE": Object {
                  "content": Object {
                    "para1": Object {
                      "classes": "mx-2",
                      "text": "You can update the contact information for your ",
                    },
                    "para2": Object {
                      "classes": "mx-2",
                      "text": "company on the User Profile page.",
                    },
                  },
                },
                "TYPE": "NOTIFICATION",
              },
              "DELETE_CONTACT": Object {
                "ADDITIONAL_ACTION": Object {
                  "action": "closeModal",
                  "classes": "mt-0 d-inline-block text-right mr-3 blue font-size-15",
                  "text": "Back",
                },
                "BODY_CLASSES": "text-center overflow-x-hidden",
                "BODY_CONTENT_CLASSES": "px-5 font-size-15",
                "FOOTER_CLASSES": "margin-unset bg-blue mt-4",
                "HEADER": "Delete Contact Information",
                "MODAL_DIALOG_CLASSES": "modal-lg",
                "PRIMARY_ACTION": Object {
                  "action": "deleteContactInfo",
                  "classes": "text-right float-right px-3",
                  "containerClasses": "p-3",
                  "text": "Delete",
                },
                "SUBTITLE": Object {
                  "classes": "px-4 mb-5",
                  "content": Object {
                    "para1": Object {
                      "classes": "mx-2 text-center mb-0 mt-4 font-size-16",
                      "text": "Please note that the contact information of the user submitting the claim will be shared with the reported seller(s) by default if you delete the public contact information.",
                    },
                  },
                },
                "TYPE": "NOTIFICATION",
              },
              "EMAIL_VERIFIED": Object {
                "BODY_CLASSES": "px-5",
                "BODY_CONTENT_CLASSES": "text-left px-5",
                "IMAGE": "ProblemSolving",
                "MESSAGE": Object {
                  "classes": "mt-4",
                  "content": Object {
                    "list": Object {
                      "liClasses": "font-size-16 mt-4",
                      "steps": Array [
                        "The information provided will be verified by our team.",
                        "You will receive a confirmation email once your application has been submitted. We will keep you updated on the status of your application via email.",
                      ],
                      "subType": "1",
                      "type": "ul",
                    },
                    "para1": Object {
                      "classes": "mx-4 font-size-16",
                      "text": "Next, we need to verify that you are the owner of the intellectual property rights that you wish to enforce.",
                    },
                    "para3": Object {
                      "classes": "font-weight-bold mt-2 mb-4 mx-4 font-size-28 text-center",
                      "text": "Your email address has been verified",
                    },
                  },
                },
                "MODAL_DIALOG_CLASSES": "width-unset modal-md",
                "PRIMARY_ACTION": Object {
                  "action": "hideModal",
                  "classes": "my-4",
                  "text": "Start",
                },
                "TYPE": "CTA",
              },
              "GO_TO_USER_PROFILE": Object {
                "ADDITIONAL_ACTION": Object {
                  "action": "updateUser",
                  "classes": "mt-0 d-inline-block text-right mr-3 blue",
                  "text": "I'll do it later",
                },
                "BODY_CLASSES": "text-center pt-5 overflow-x-hidden",
                "BODY_CONTENT_CLASSES": "px-5 py-2 font-size-18",
                "DISPLAY_DASHBOARD": true,
                "FOOTER_CLASSES": "margin-unset bg-blue mt-4",
                "HEADER": "Welcome to Walmart Brand Portal!",
                "HEIGHT": "110",
                "IMAGE": "ProblemSolving",
                "PRIMARY_ACTION": Object {
                  "action": "reroute",
                  "actionParam": "/profile",
                  "classes": "text-right float-right px-3",
                  "containerClasses": "p-3",
                  "text": "Go to User Profile",
                },
                "SUBTITLE": Object {
                  "content": Object {
                    "para1": Object {
                      "classes": "mx-2 font-size-16 py-4 px-5",
                      "text": "You can update the public contact information on the User Profile page. This contact information will be shared with sellers reported by the user(s) of this Brand Portal account.",
                    },
                  },
                },
                "TYPE": "NOTIFICATION",
              },
              "LOGOUT": Object {
                "ADDITIONAL_ACTION": Object {
                  "action": "hideModal",
                  "text": "Go Back",
                },
                "BODY_CLASSES": "px-5",
                "BODY_CONTENT_CLASSES": "text-left",
                "HEIGHT": 100,
                "MESSAGE": Object {
                  "classes": "",
                  "content": Object {},
                },
                "MODAL_DIALOG_CLASSES": "width-unset",
                "PRIMARY_ACTION": Object {
                  "action": "logout",
                  "text": "Logout",
                },
                "SUBTITLE": Object {
                  "classes": "mx-4",
                  "content": Object {
                    "para1": Object {
                      "classes": "font-size-16 mx-5",
                      "text": "You will lose your progress and the information already entered will not be saved.",
                    },
                  },
                },
                "TITLE": Object {
                  "content": Object {
                    "para": Object {
                      "classes": "h3 mx-auto mb-2 font-weight-bold",
                      "text": "Are you sure you want to logout?",
                    },
                  },
                },
                "TYPE": "CTA",
              },
              "PASSWORD_RESET_SELLER": Object {
                "ADDITIONAL_ACTION": Object {
                  "action": "closeModal",
                  "classes": "mt-0 font-size-15",
                  "containerClasses": "mr-4 d-inline-block",
                  "text": "Cancel",
                },
                "BODY_CLASSES": "overflow-x-hidden",
                "BODY_CONTENT_CLASSES": "px-3 font-size-15",
                "DISPLAY_DASHBOARD": true,
                "FOOTER_CLASSES": "margin-unset bg-blue mt-4",
                "HEADER": "Seller Center Profile",
                "MESSAGE": Object {
                  "classes": "mt-4",
                  "content": Object {
                    "para1": Object {
                      "classes": "mx-2",
                      "text": "To change your Seller Center credentials, you will need to log into your Seller Center account.",
                    },
                    "para2": Object {
                      "classes": "mx-2 mb-5",
                      "text": "You cannot make changes to your Seller Center account through the Brand Portal.",
                    },
                  },
                },
                "PRIMARY_ACTION": Object {
                  "action": "navigation",
                  "actionParam": "https://seller.stg.walmart.com",
                  "classes": "text-right float-right px-3",
                  "containerClasses": "p-3",
                  "text": "Go to Seller Center",
                },
                "TYPE": "NOTIFICATION",
              },
              "PORTAL_DASHBOARD": Object {
                "CODE": 4,
              },
              "PORTAL_REGISTRATION": Object {
                "CODE": 1,
              },
              "REQUEST_ACCESS": Object {
                "CODE": 32,
                "IMAGE": "GreenCircleCheck",
                "MESSAGE": "Access Request, awaiting approval.",
                "TITLE": "Access requested",
              },
              "SELLER_WELCOME_PROMPT": Object {
                "BODY_CLASSES": "px-5",
                "BODY_CONTENT_CLASSES": "text-left px-5",
                "IMAGE": "ProblemSolving",
                "MESSAGE": Object {
                  "classes": "mt-4",
                  "content": Object {
                    "para1": Object {
                      "classes": "font-weight-bold mt-2 mb-4 mx-4 font-size-28 text-center",
                      "dynamicReplacementConfig": Object {
                        "__firstNamePlaceholder__": "profile.firstName",
                        "__lastNamePlaceholder__": "profile.lastName",
                      },
                      "text": "Welcome __firstNamePlaceholder__ __lastNamePlaceholder__",
                    },
                    "para2": Object {
                      "classes": "mx-3 font-size-16 text-center",
                      "text": "We will be using the information in your Seller Center account to set up your Brand Portal account.",
                    },
                    "para3": Object {
                      "classes": "mx-3 mt-5 font-size-16 text-center",
                      "text": "Please confirm the information of your company and provide the details of your registered trademark.",
                    },
                    "para4": Object {
                      "classes": "mx-4 mt-5 font-size-16 text-center",
                      "text": "The information provided will be verified by our team. You will receive a confirmation email once your application status has been updated.",
                    },
                  },
                },
                "MODAL_DIALOG_CLASSES": "width-unset modal-md",
                "PRIMARY_ACTION": Object {
                  "action": "hideModal",
                  "classes": "my-4",
                  "text": "Start",
                },
                "TYPE": "CTA",
              },
              "TOU_VERIFICATION": Object {
                "CODE": 128,
                "FILE": "test.pdf",
                "GENERIC": true,
                "TEMPLATE": "TouTemplate",
              },
              "USER_ACCESS_REVOKED": Object {
                "CODE": 16,
                "IMAGE": "RedCircleCross",
                "MESSAGE": "Your account access has been revoked for security reasons",
                "TITLE": Object {
                  "ROW_CLASSES": "justify-content-center",
                  "content": Object {
                    "para": Object {
                      "classes": "font-weight-bold",
                      "text": "Access Denied",
                    },
                  },
                },
              },
              "USER_VERIFICATION": Object {
                "ADDITIONAL_ACTION": Object {
                  "action": "resendInvite",
                  "actionHelpText": Object {
                    "classes": "font-size-14",
                    "text": "No email received?",
                  },
                  "text": "Resend Verification Email",
                },
                "CODE": 64,
                "HEIGHT": "120",
                "IMAGE": "EmailIconWithYellowMessage",
                "MESSAGE": Object {
                  "classes": "background-lightblue mx-2 my-3 px-5 py-3 font-size-16",
                  "content": Object {
                    "partial1": Object {
                      "chunk1": "We've sent a verification email to ",
                      "chunk2": Object {
                        "classes": "font-weight-bold",
                        "dynamicReplacementConfig": Object {
                          "__emailPlaceholder__": "profile.email",
                        },
                        "text": "__emailPlaceholder__",
                      },
                      "chunk3": ". Follow the link in the email to complete verification.",
                    },
                  },
                },
                "MODAL_DIALOG_CLASSES": "width-medium",
                "NO_PRIMARY_ACTION": true,
                "SUBTITLE": Object {
                  "content": Object {
                    "para2": Object {
                      "classes": "font-size-15 mx-5",
                      "text": "To move forward with your application, we need to verify the email address associated with your account.",
                    },
                  },
                },
                "TITLE": Object {
                  "ROW_CLASSES": "justify-content-center",
                  "content": Object {
                    "para": Object {
                      "classes": "h5 mx-5 px-3 mb-4 font-weight-bold",
                      "text": "Please verify your email address",
                    },
                  },
                },
              },
            }
          }
          updateUserProfile={[Function]}
          userProfile={
            Object {
              "brands": Array [
                Object {
                  "id": "640a20c2-3bbd-46e5-9a81-4f97c8bc9f08",
                },
              ],
              "context": "edit",
              "createTs": "2020-09-15T07:15:18.965Z",
              "createdBy": "wm.ropro.stg+demo@gmail.com",
              "email": "wm.ropro.stg+demo@gmail.com",
              "emailVerified": true,
              "firstName": "Test Mohit",
              "isOrgEnabled": true,
              "isUserEnabled": true,
              "lastName": "Nagar",
              "lastUpdateTs": "2020-09-21T10:06:07.633Z",
              "lastUpdatedBy": "wm.ropro.stg+demo@gmail.com",
              "organization": Object {
                "id": "640a20c2-3bbd-46e5-9a81-4f97c8bc9f08",
                "status": "Accepted",
              },
              "phoneCountry": "1",
              "phoneNumber": "(234) 567-8901",
              "registrationMode": "SelfRegistered",
              "role": Object {
                "description": "Brand Rights Owner",
                "id": "6a429471-3675-4490-93db-5aadf5412a8b",
                "name": "Super Admin",
              },
              "sellerInfo": Object {
                "authTenantId": "Walmart_MP_US",
                "bizRole": "SELLER",
                "buId": "0",
                "businessName": "StageTestSeller 2@DD",
                "countryOfIncorporation": "CN",
                "email": "ss1@mailinator.com",
                "firstName": "Shakuntala",
                "isPreferredMart": "true",
                "lastName": "Sureban",
                "legalName": "Test Name",
                "martId": "0",
                "orgId": "30b86e9e-a7f6-4d62-989f-4937a8e79d8c",
                "orgStatus": "NO_STATUS",
                "organizationAddress": Object {
                  "address1": "8B tinglan lane shahu technology park",
                  "city": "Dongcheng",
                  "country": "CN",
                  "postalcode": "100009",
                  "state": "BJ",
                },
                "partnerId": "2857090a-a4bb-4bc3-a959-6aa3f2dd8d17",
                "principalId": "e6a3b2bd-a6c9-44fa-ab8b-7fbfb280c464",
                "realmId": "30b86e9e-a7f6-4d62-989f-4937a8e79d8c",
                "taxClassification": "W-8 BEN-E",
              },
              "status": "Active",
              "statusDetails": "Status updated by: system",
              "type": "Internal",
              "workflow": Object {
                "code": 4,
                "defaultView": "portal-view-users",
                "roleCode": 1,
                "roleView": "SUPER_ADMIN",
                "workflow": "portal_dashboard",
              },
            }
          }
        >
          <div
            className="fill-parent loader"
          />
        </Authenticator>
      </Connect(Authenticator)>
    </Router>
  </BrowserRouter>
</Provider>
`;
